x-pack/platform/plugins/shared/ml/public/application/model_management/hooks/use_init_trained_models_service.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useMemo } from 'react';
+import { BehaviorSubject } from 'rxjs';
+import { useStorage } from '@kbn/ml-local-storage';
+import { ML_ACTIVE_MODEL_DEPLOYMENTS } from '../../../../common/types/storage';
+import type { ModelDeploymentParams } from '../trained_models_service';
+import type { TrainedModelsService } from '../trained_models_service';
+import { useMlKibana } from '../../contexts/kibana';
+import { useToastNotificationService } from '../../services/toast_notification_service';
+
+/**
+ * Hook that initializes the shared TrainedModelsService instance with storage
+ * for tracking active operations. The service is destroyed when no components
+ * are using it and all operations are complete.
+ */
+export function useInitTrainedModelsService(): TrainedModelsService {
+  const {
+    services: {
+      mlServices: { trainedModelsService },
+    },
+  } = useMlKibana();
+
+  const { displayErrorToast, displaySuccessToast } = useToastNotificationService();
+
+  const initialDeployingState = useMemo(() => [], []);
+
+  const [deployingModels, setDeployingModels] = useStorage<
+    typeof ML_ACTIVE_MODEL_DEPLOYMENTS,
+    ModelDeploymentParams[]
+  >(ML_ACTIVE_MODEL_DEPLOYMENTS, initialDeployingState);
+
+  const deployingModels$ = useMemo(
```

Comment:
This observable contains not currently deploying models, but the scheduled ones
```suggestion
  const scheduledDeployments$ = useMemo(
```

Reply:
Renamed in https://github.com/elastic/kibana/pull/205699/commits/e88fd196e0ef0d97160702a507515c570429bb05